plugins/revealer/qt.py

```diff
@@ -0,0 +1,688 @@
+
+from electrum.plugins import BasePlugin, hook
+from electrum.i18n import _
+from electrum_gui.qt.util import *
+from electrum_gui.qt.qrtextedit import ScanQRTextEdit
+from electrum.util import to_bytes
+
+
+class Plugin(BasePlugin):
+
+    def __init__(self, parent, config, name):
+        BasePlugin.__init__(self, parent, config, name)
+        self.base_dir = config.electrum_path()+'/revealer/'
+
+        if self.config.get('calibration_h') == None:
+            self.config.set_key('calibration_h', 0)
+        if self.config.get('calibration_v') == None:
+            self.config.set_key('calibration_v', 0)
+
+        self.calibration_h = self.config.get('calibration_h')
+        self.calibration_v = self.config.get('calibration_v')
+
+        self.version = '0'#
+        self.size = (159, 97)
+        self.f_size = QSize(1014*2, 642*2)
+        self.abstand_h = 21
+        self.abstand_v = 34
+        self.calibration_noise = int('10' * 128)
+        self.rawnoise = False
+        if not os.path.exists(self.base_dir):
+            os.mkdir(self.base_dir)
+
+    @hook
+    def set_seed(self, seed, parent):
+        self.cseed = seed.upper()
+        parent.addButton(':icons/revealer.png', partial(self.setup_dialog, parent), "Revealer"+_("secret backup utility"))
+
+    def requires_settings(self):
+        return True
+
+    def settings_widget(self, window):
+        return EnterButton(_('Printer Calibration'), partial(self.calibration_dialog, window))
+
+    def setup_dialog(self, window):
+        self.update_wallet_name(window.parent().parent().wallet)
+        self.user_input = False
+        self.noise_seed = False
+        self.d = WindowModalDialog(window, "Revealer")
+        self.d.setMinimumWidth(420)
+        vbox = QVBoxLayout(self.d)
+        vbox.addSpacing(21)
+        logo = QLabel()
+        vbox.addWidget(logo)
+        logo.setPixmap(QPixmap(':icons/revealer.png'))
+        logo.setAlignment(Qt.AlignCenter)
+        vbox.addSpacing(42)
+
+        self.load_noise = ScanQRTextEdit()
+        self.load_noise.setTabChangesFocus(True)
+        self.load_noise.textChanged.connect(self.on_edit)
+        self.load_noise.setMaximumHeight(33)
+
+        vbox.addWidget(WWLabel("<b>"+_("Enter your physical revealer code:")+"<b>"))
+        vbox.addWidget(self.load_noise)
+        vbox.addSpacing(11)
+
+        self.next_button = QPushButton(_("Next"), self.d)
+        self.next_button.setDefault(True)
+        self.next_button.setEnabled(False)
+        vbox.addLayout(Buttons(self.next_button))
+        self.next_button.clicked.connect(self.d.close)
+        self.next_button.clicked.connect(partial(self.cypherseed_dialog, window))
+        vbox.addSpacing(21)
+
+        vbox.addWidget(WWLabel(_("or, alternatively: ")))
+        bcreate = QPushButton(_("Create a digital Revealer"))
+        bcreate.clicked.connect(partial(self.make_digital, self.d))
+        bcreate.clicked.connect(partial(self.cypherseed_dialog, window))
```

Comment:
The second button callback is actually dependent on the first, isn't it?
You should create a method (or inner function) that calls the first in a try-catch, and only calls the second if there was no exception.